src/components/overlay/overlay.ts

```diff
@@ -0,0 +1,376 @@
+import { html, unsafeStatic } from 'lit/static-html.js';
+
+import { FocusHandler, getFirstFocusableElement, setModalityOnNextFocus } from '../core/a11y';
+import { LanguageController, SbbNegativeMixin } from '../core/common-behaviors';
+import { ScrollHandler, isValidAttribute, hostContext } from '../core/dom';
+import { EventEmitter } from '../core/eventing';
+import { i18nCloseDialog, i18nDialog, i18nGoBack } from '../core/i18n';
+import type { SbbOverlayState } from '../core/overlay';
+import { applyInertMechanism, removeInertMechanism } from '../core/overlay';
+
+import style from './overlay.scss?lit&inline';
+
+// A global collection of existing overlays
+const overlayRefs: SbbOverlayElement[] = [];
+let nextId = 0;
+
+type CloseEventDetails = {
+  returnValue?: any;
+  closeTarget?: HTMLElement;
+};
+
+/**
+ * It displays an interactive overlay element.
+ *
+ * @slot - Use the unnamed slot to provide a content for the overlay.
+ * @event {CustomEvent<void>} willOpen - Emits whenever the `sbb-overlay` starts the opening transition. Can be canceled.
+ * @event {CustomEvent<void>} didOpen - Emits whenever the `sbb-overlay` is opened.
+ * @event {CustomEvent<void>} willClose - Emits whenever the `sbb-overlay` begins the closing transition. Can be canceled.
+ * @event {CustomEvent<CloseEventDetails>} didClose - Emits whenever the `sbb-overlay` is closed.
+ * @event {CustomEvent<void>} requestBackAction - Emits whenever the back button is clicked.
+ * @cssprop [--sbb-overlay-z-index=var(--sbb-overlay-z-index)] - To specify a custom stack order,
+ * the `z-index` can be overridden by defining this CSS variable. The default `z-index` of the
+ * component is set to `var(--sbb-overlay-z-index)` with a value of `1000`.
+ */
+@customElement('sbb-overlay')
+export class SbbOverlayElement extends SbbNegativeMixin(LitElement) {
+  public static override styles: CSSResultGroup = style;
+  public static readonly events = {
+    willOpen: 'willOpen',
+    didOpen: 'didOpen',
+    willClose: 'willClose',
+    didClose: 'didClose',
+    backClick: 'requestBackAction',
+  } as const;
+
+  /**
+   * Whether to allow the overlay content to stretch to full width.
+   * By default, the content has the appropriate page size.
+   */
+  @property({ reflect: true, type: Boolean }) public expanded = false;
+
+  /**
+   * Whether a back button is displayed next to the title.
+   */
+  @property({ attribute: 'back-button', type: Boolean }) public backButton = false;
+
+  /**
+   * This will be forwarded as aria-label to the close button element.
+   */
+  @property({ attribute: 'accessibility-close-label' }) public accessibilityCloseLabel:
+    | string
+    | undefined;
+
+  /**
+   * This will be forwarded as aria-label to the back button element.
+   */
+  @property({ attribute: 'accessibility-back-label' }) public accessibilityBackLabel:
+    | string
+    | undefined;
+
+  /**
+   * This will be forwarded as aria-label to the relevant nested element.
```

Comment:
maybe we should more specifically describe what will be described with this label